src/reducer.ts

```diff
@@ -162,14 +172,14 @@ type ReducerCreator = {
 };
 
 export const reducer: ReducerCreator = <State>(
-  trigger: UnknownActionCreator | UnknownActionCreator[] | Observable<any>,
+  trigger: UnknownActionCreator | UnknownActionCreator[] | ObservableInput<any>,
   reducerFn: Reducer<State, any>
 ) => {
   const wrapper = (state: State, payload: any, namespace?: string) =>
     reducerFn(state, payload, namespace);
-  if (trigger instanceof Observable) {
+  if (!Array.isArray(trigger) && isObservableInput(trigger)) {
```

Comment:
This doesn't really sit well with me, but I think we can fix it after this release

Reply:
As we discussed, I think from an interface and readability point of view it would be better to have a `actionReducer` and `stateReducer`, that would probably also allow to make the code here a bit simpler.